sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/SchemaIOTableProviderWrapper.java

```diff
@@ -80,13 +80,17 @@ public BeamSqlTable buildBeamSqlTable(Table tableDefinition) {
     }
   }
 
-  private BeamTableStatistics getTableStatistics(PipelineOptions options) {
+  public BeamTableStatistics getTableStatistics(PipelineOptions options) {
     if (isBounded().equals(PCollection.IsBounded.BOUNDED)) {
       return BeamTableStatistics.BOUNDED_UNKNOWN;
     }
     return BeamTableStatistics.UNBOUNDED_UNKNOWN;
   }
 
+  public BeamTableStatistics getTableStatistics(PipelineOptions options, SchemaIO schemaIO) {
+    return getTableStatistics(options);
+  }
+
```

Comment:
I don't think this new method is actually called anywhere. Should it be called from `SchemaIOTableWrapper#getTableStatistics`?

 It might also be good to just get rid of the other `getTableStatistics` and move the default implementation to this one.

Reply:
The reason why I added this additional getTableStatistics() function was that DataStoreV1 is the first IO whose getTableStatistics() function relied on the schemaIO data, and this SchemaIOTableWrapper#getTableStatistics function could be overridden in DataStoreV1TableProvider. The other IOs so far have not required the schemaIO. Do you think we should get rid of the other getTableStatistics() function and always require a schemaIO, even when the schemaIO isn't absolutely necessary? Or support both cases?

Reply:
I see what you mean now. I got rid of the old getTableStatistics() function and moved the default implementation to this one.